### ISSUE TITLE
Loss Rise due to Gradient Explosion

### ISSUE THREAD

**Issue by Yumenomae** [Yumenomae/PASSAT_Physics-assisted_and_topology-informed_Weather_Prediction#2]
Hi! We have observed that PASSAT may occasionally experience a sharp increase in the loss function during training, often due to gradient explosion. This phenomenon appears to be linked to the interaction between the intermediate results of solving the Navier-Stokes and advection equations and the neural network itself. In our training logs, we have noted sudden increases in the loss, such as from 0.03 to 0.20. If you encounter a similar issue, we strongly recommend adjusting the `CLIP_GRAD` parameter, then resuming. Specifically, lowering the value from 1.0 to a range between 0.1 and 0.3 can help mitigate the gradient explosion, stabilizing the training process.

**Comment by Yumenomae** [comment]
For your reference, we have attached our training log, which includes instances of restarts triggered by gradient explosions. Notably, you’ll see adjustments made to the `CLIP_GRAD` parameter, particularly around the 11th~13th epoch, where it was reduced from 1.0 to 0.3. We hope this provides additional context to better understand how these changes impacted the stability of the training process. 
We do not anticipate that adjusting the `CLIP_GRAD` parameter for PASSAT will affect the fairness of the comparison. Lowering the `CLIP_GRAD` value for other models, however, would result in a reduction of their performance, potentially skewing the comparison. Therefore, we believe that this adjustment is specific to PASSAT's unique behavior and does not introduce any unfair bias in the results.
[log_rank0.txt](https://github.com/user-attachments/files/20228969/log_rank0.txt)
